### DISCUSSION TITLE
QUESTION: Logic used to determine contents of swap file

### DISCUSSION THREAD

**Discussion by cshanahan** [vim/vim#15870] [Q&A]
QUESTION:
Can someone point me to the place(s) in the code or the logic that is responsible for determining the data which is added to the swap file when a large text file is first opened. For example, when a large text file (e.g. more than 10k lines) is opened, but before anything is done with the file, some amount of data is added to the swap file. Data from the start of the file and data from the end of the file. What factors or logic is used to determine this?

When I open a large text file, a 4k swap file is created. After a few seconds the swap file changes to 16k, and it contains data from the text file -- log lines from the beginning of the file and log lines from the end of the file. This occurs before any editing or commands in Vim.

I have scanned through the codebase already and nothing stands out to me; I am not familiar enough with the codebase.
I have also read many docs so I am well aware of the purpose of the swap file, and generally what contents can be found in the swap file. I cannot seem to find the logic which explains what data from an opened file is added to the swap file at the start of a session.

I'm sure it doesn't matter, but I am using Vim v9.1 on openSUSE Leap 15.6.

TY

**Comment by zeertzjq** [comment]
Mostly in src/memfile.c
